File: C++/Q30.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
	long long n;
	cout << "Enter a positive integer: ";
	if (!(cin >> n) || n <= 0) {
		cout << "Invalid input. Enter a positive integer." << endl;
		return 1;
	}
	cout << "Factors of " << n << " are: ";
	bool first = true;
	for (long long i = 1; i * i <= n; ++i) {
		if (n % i == 0) {
			if (!first) cout << ' ';
			cout << i;
			first = false;
			if (i != n / i) cout << ' ' << (n / i);
		}
	}
	cout << endl;
	return 0;
}

File: Java/Q30.java
import java.util.Scanner;

public class FactorsOfNumber {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Input from user
        System.out.print("Enter a number: ");
        int number = scanner.nextInt();

        System.out.println("Factors of " + number + " are:");

        // Loop to find factors
        for (int i = 1; i <= number; i++) {
            if (number % i == 0) {
                System.out.print(i + " ");
            }
        }

        scanner.close();
    }
}